extern_prelude check is breaking rustdoc

See https://github.com/rust-lang-nursery/futures-rs/pull/1003#issuecomment-387658833

```
 Documenting futures-channel v0.2.1 (file:///home/travis/build/rust-lang-nursery/futures-rs/futures-channel)
error[E0658]: access to extern crates through prelude is experimental (see issue #44660)
  |
  = help: add #![feature(extern_prelude)] to the crate attributes to enable
error: Could not document `futures-channel`.
Caused by:
  process didn't exit successfully: `rustdoc --crate-name futures_channel futures-channel/src/lib.rs -o /home/travis/build/rust-lang-nursery/futures-rs/target/doc --cfg feature="default" --cfg feature="futures-core" --cfg feature="std" -L dependency=/home/travis/build/rust-lang-nursery/futures-rs/target/debug/deps --extern futures_core=/home/travis/build/rust-lang-nursery/futures-rs/target/debug/deps/libfutures_core-c5293a3c307e0c7b.rmeta` (exit code: 101)
warning: build failed, waiting for other jobs to finish...
```

--- a/src/librustc/session/config.rs
+++ b/src/librustc/session/config.rs
@@ -494,6 +494,13 @@ impl Input {
             Input::Str { .. } => "rust_out".to_string(),
         }
     }
+
+    pub fn get_input(&mut self) -> Option<&mut String> {
+        match *self {
+            Input::File(_) => None,
+            Input::Str { ref mut input, .. } => Some(input),
+        }
+    }
 }
 
 #[derive(Clone)]

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -1437,6 +1437,9 @@ pub struct Resolver<'a> {
     current_type_ascription: Vec<Span>,
 
     injected_crate: Option<Module<'a>>,
+
+    /// Only supposed to be used by rustdoc, otherwise should be false.
+    pub ignore_extern_prelude_feature: bool,
 }
 
 /// Nothing really interesting here, it just provides memory for the rest of the crate.
@@ -1718,6 +1721,7 @@ impl<'a> Resolver<'a> {
             unused_macros: FxHashSet(),
             current_type_ascription: Vec::new(),
             injected_crate: None,
+            ignore_extern_prelude_feature: false,
         }
     }
 
@@ -1891,7 +1895,8 @@ impl<'a> Resolver<'a> {
         if !module.no_implicit_prelude {
             // `record_used` means that we don't try to load crates during speculative resolution
             if record_used && ns == TypeNS && self.extern_prelude.contains(&ident.name) {
-                if !self.session.features_untracked().extern_prelude {
+                if !self.session.features_untracked().extern_prelude &&
+                   !self.ignore_extern_prelude_feature {
                     feature_err(&self.session.parse_sess, "extern_prelude",
                                 ident.span, GateIssue::Language,
                                 "access to extern crates through prelude is experimental").emit();

--- a/src/librustdoc/clean/mod.rs
+++ b/src/librustdoc/clean/mod.rs
@@ -1103,7 +1103,7 @@ fn resolve(cx: &DocContext, path_str: &str, is_val: bool) -> Result<(Def, Option
             // early return and try looking for the trait
             let value = match result.def {
                 Def::Method(_) | Def::AssociatedConst(_) => true,
-                Def::AssociatedTy(_)  => false,
+                Def::AssociatedTy(_) => false,
                 Def::Variant(_) => return handle_variant(cx, result.def),
                 // not a trait item, just return what we found
                 _ => return Ok((result.def, None))

--- a/src/librustdoc/core.rs
+++ b/src/librustdoc/core.rs
@@ -250,9 +250,11 @@ pub fn run_core(search_paths: SearchPaths,
                                                         |_| Ok(()));
         let driver::InnerExpansionResult {
             mut hir_forest,
-            resolver,
+            mut resolver,
             ..
         } = abort_on_err(result, &sess);
+
+        resolver.ignore_extern_prelude_feature = true;
 
         // We need to hold on to the complete resolver, so we clone everything
         // for the analysis passes to use. Suboptimal, but necessary in the

